### ISSUE TITLE
Resolved spam messages should be sent to their intended recipients

### ISSUE THREAD

**Issue by tiwane** [inaturalist/inaturalist#3221]
Currently if the system flags a message as spam and that flag is resolved, the message is never sent to its intended recipient. If a spam message is resolved, we should send it to the intended recipient and generate a notification (including an email one, if applicable).

**Comment by pleary** [comment]
Could you describe a recent case of this @tiwane? What might be happening is a user creates a message which is flagged as spam, and _also_ marks the user as a spammer. When a user is marked as a spammer any [unread messages they have sent are deleted](https://github.com/inaturalist/inaturalist/blob/main/app/models/user.rb#L1276). At that point whether the flag is resolved and/or the user is unsuspended and marked as a non-spammer, the original message will not be visible by the original recipient because it was deleted. The only case I can see right now where the recipient wouldn't see the message is if the sending user were suspended.

If that's the case that the messages are lost when and only when a user is suspended, what would be the best course of action? We might be able to detect which messages the user sent but were not received by the recipient and try sending them again.

If the user is unsuspended but the message is still flagged, perhaps we don't want to send it again. And then if the user is unsuspended while the message is flagged, and the message is later unflagged, that may be another time we want to try sending it again if the recipient hasn't received it.

If the message flag is resolved while the user is still suspended, we also probably don't want to send it again.

Does that seem like a plausible explanation: that this only happens when the flagged message causes the sending user to be suspended? Any recommendations on remediation? Maybe we could attempt to resend at the time when both the user is unsuspended and the message flag is resolved, and the recipient has not already received the message.

**Comment by tiwane** [comment]
> What might be happening is a user creates a message which is flagged as spam, and also marks the user as a spammer.

Yes, that's usually the case. New user, usually a researcher, sends a few messages asking for details and those get flagged as spam. Once three messages are flagged as spam, the system flags the account as a spammer. I usually just flag them as a non-spammer, which reinstates their account and should resolve the spam flags. Here's an example: https://inaturalist.slack.com/archives/C04MF2V4ZAN/p1700683463422049

> Maybe we could attempt to resend at the time when both the user is unsuspended and the message flag is resolved, and the recipient has not already received the message.

This sounds good to me.
